src/descriptor/dsl.rs

```diff
@@ -640,21 +764,22 @@ macro_rules! fragment {
             .and_then(|items| $crate::fragment!(thresh_vec($thresh, items)))
     });
     ( multi_vec ( $thresh:expr, $keys:expr ) ) => ({
-        $crate::keys::make_multi($thresh, $keys)
+        let secp = $crate::bitcoin::secp256k1::Secp256k1::new();
+
+        $crate::keys::make_multi($thresh, $crate::miniscript::Terminal::Multi, $keys, &secp)
     });
     ( multi ( $thresh:expr $(, $key:expr )+ ) ) => ({
-        use $crate::keys::IntoDescriptorKey;
+        $crate::group_multi_keys!( $( $key ),* )
```

Comment:
Probably I am missing something, but why the input `key` pattern is `1 or many` and the pattern in code is `0 or many`? (the `+` in input and `*` in the code after `key`)

Reply:
I think it doesn't matter what you use when expanding repeated items, the compiler only looks at what you have in the "declaration" to match the tokens passed to the macro.

So in this case it should error if there's no key at all (because of the `+`), but afterwards you just want to iterate over whatever you have and for that you can just use `*`.